state/txindex/kv/kv.go

```diff
@@ -299,43 +312,20 @@ func startKeyForRange(r queryRange, height uint64, heightSpecified bool) []byte
 		}
 	}
 	var key string
-	if heightSpecified {
+	if height > 0 {
 		key = fmt.Sprintf("%s/%v/%d", r.key, lowerBound, height)
 	} else {
 		key = fmt.Sprintf("%s/%v", r.key, lowerBound)
 	}
 	return []byte(key)
 }
 
-func (txi *TxIndex) matchRange(r queryRange, startKey []byte) (hashes [][]byte) {
-	it := txi.store.IteratorPrefix(startKey)
-	defer it.Release()
-	for it.Next() {
-		// no other way to stop iterator other than checking for upperBound
-		switch (r.upperBound).(type) {
-		case int64:
-			v, err := strconv.ParseInt(extractValueFromKey(it.Key()), 10, 64)
-			if err == nil && v == r.upperBound {
-				if r.includeUpperBound {
-					hashes = append(hashes, it.Value())
-				}
-				break
-			}
-			// XXX: passing time in a ABCI Tags is not yet implemented
-			// case time.Time:
-			// 	v := strconv.ParseInt(extractValueFromKey(it.Key()), 10, 64)
-			// 	if v == r.upperBound {
-			// 		break
-			// 	}
-		}
-		hashes = append(hashes, it.Value())
-	}
-	return
+func isTagKey(key []byte) bool {
+	return strings.Count(string(key), tagKeySeparator) == 3
 }
 
 func extractValueFromKey(key []byte) string {
-	s := string(key)
-	parts := strings.SplitN(s, "/", 3)
+	parts := strings.SplitN(string(key), tagKeySeparator, 3)
 	return parts[1]
```

Comment:
I see that this is safe but would be helpful to code it defensively or clarify why